python/ray/workflow/task_executor.py

```diff
@@ -68,10 +69,21 @@ def _workflow_task_executor(
     """
     with workflow_context.workflow_task_context(context):
         store = workflow_storage.get_workflow_storage()
+        execution_metadata = WorkflowExecutionMetadata(
+            node_id=ray.get_runtime_context().get_node_id()
+        )
+
         # Part 1: resolve inputs
         args, kwargs = baked_inputs.resolve(store)
 
-        # Part 2: execute the task
+        # Part 2: block util upstream checkpoint tasks completes successfully.
+        try:
+            baked_inputs.wait_upstream_checkpoint_tasks()
+        except RayError:
+            execution_metadata.upstream_checkpointing_failed = True
+            return execution_metadata, None
```

Comment:
This should not be an issue, because

1. previously, when checkpoint inside a task, it blocks the previous task and holding more resources
2. ray will release blocked resources when a task is blocked by `ray.get`

Reply:
> previously, when checkpoint inside a task, it blocks the previous task and holding more resources

I think we are not launching the new task before the checkpointing finished right? Am I missing something?

Reply:
yes, but it blocks the previous task, and the previous task still holds the resources

Reply:
Got it!